prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/git/GitCloneTask.java

```diff
@@ -0,0 +1,82 @@
+
+import com.redhat.parodos.workflow.context.WorkContextDelegate;
+import com.redhat.parodos.workflow.exception.MissingParameterException;
+import com.redhat.parodos.workflow.parameter.WorkParameter;
+import com.redhat.parodos.workflow.parameter.WorkParameterType;
+import com.redhat.parodos.workflow.task.BaseWorkFlowTask;
+import com.redhat.parodos.workflows.work.DefaultWorkReport;
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.api.errors.InvalidRemoteException;
+import org.eclipse.jgit.api.errors.TransportException;
+
+@AllArgsConstructor
+public class GitCloneTask extends BaseWorkFlowTask {
+
+	@Override
+	public @NonNull List<WorkParameter> getWorkFlowTaskParameters() {
+		return List.of(
+				WorkParameter.builder().key(GitUtils.getUri()).type(WorkParameterType.TEXT).optional(false)
+						.description("Url to clone from").build(),
+				WorkParameter.builder().key(GitUtils.getBranch()).type(WorkParameterType.TEXT).optional(true)
+						.description("Branch to clone from, default main").build(),
+				WorkParameter.builder().key("credentials").type(WorkParameterType.TEXT).optional(false)
+						.description("Git credential").build());
+	}
+
+	@Override
+	public WorkReport execute(WorkContext workContext) {
+		String gitUri = null;
+		String destination = null;
+		String gitBranch = null;
+
+		try {
+			gitUri = WorkContextDelegate.getRequiredValueFromRequestParams(workContext, GitUtils.getUri());
+			gitBranch = WorkContextDelegate.getOptionalValueFromRequestParams(workContext, GitUtils.getBranch(),
+					"main");
+			destination = cloneRepo(gitUri, gitBranch);
+		}
+		catch (MissingParameterException e) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, e);
+		}
+		catch (TransportException e) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext,
+					new Exception("cannot connect to the repository server"));
+		}
+		catch (InvalidRemoteException e) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext,
+					new Exception("Remote repository " + gitUri + " is not available"));
+		}
+		catch (IOException | GitAPIException e) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext,
+					new Exception("cannot clone repository, error: " + e.getMessage()));
+		}
+
+		workContext.put(GitUtils.getContextUri(), gitUri);
+		workContext.put(GitUtils.getContextDestination(), destination);
+		workContext.put(GitUtils.getContextBranch(), gitBranch);
+		return new DefaultWorkReport(WorkStatus.COMPLETED, workContext, null);
+	}
+
+	private String cloneRepo(String gitUri, String gitBranch)
+			throws InvalidRemoteException, TransportException, IOException, GitAPIException {
+		Path tmpPath = (Path) Paths.get("/tmp/");
```

Comment:
is there a need to specify the absolute path before calling `Files.createTempDirector`?
there is `createTempDirectory` without a prefix that can be used. Asking to know why we should point on `/tmp` and not let java determines the default temp folder.

Reply:
Since @gciavarrini suggested to use this function, Is the way that I find, we need to specify the folder

Reply:
ack

Reply:
I'm not sure what do you want to change here TBH